### PR TITLE
Fix HTTP status 500.

### PR DESCRIPTION
**Pull request by jonashendrickx**

### CODE REVIEW

File: src/Api/Middleware/EventLogContextMiddleware.cs [hunk 1]
```diff
@@ -13,7 +13,14 @@ public class EventLogContextMiddleware
 
     public async Task InvokeAsync(HttpContext context, IEventLogContext eventLogContext, IFeatureContextProvider featureContextProvider)
     {
+        if (!context.Items.ContainsKey("__AuthorizationMiddlewareWithEndpointInvoked"))
```
Comment (abergs) [inline src/Api/Middleware/EventLogContextMiddleware.cs:16]: Would benefit from a comment why this is done. Is there any other "public" way to check if this is an authorized endpoint, `__AuthorizationMiddlewareWithEndpointInvoked` feels like a magic internal string. But I may be wrong?

Comment (jonashendrickx) [reply src/Api/Middleware/EventLogContextMiddleware.cs:19]: @abergs  see comment :-) !

File: src/Api/Middleware/EventLogStorageCommitMiddleware.cs [hunk 2]
```diff
@@ -28,7 +29,11 @@ public async Task InvokeAsync(HttpContext context, IServiceProvider provider)
         {
             try
             {
-                await provider.GetRequiredService<IEventLogger>().FlushAsync();
+                var eventLoggerContext = provider.GetRequiredService<IEventLogContext>();
+                if (!string.IsNullOrEmpty(eventLoggerContext.TenantId))
```
Comment (jrmccannon) [inline src/Api/Middleware/EventLogStorageCommitMiddleware.cs:33]: This breaks all event logging from an unauthenticated standpoint. I wrote some tests to validate those instances and can add them to this branch.

**Review by jrmccannon — CHANGES_REQUESTED** [review]
The requirement of the eventLoggerContext to exist won't work because it isn't invoked when the user hasn't been authenticated yet.

**Pull request closed without merging**